### PR TITLE
New CONTRIBUTING file

### PR DESCRIPTION
**Pull request by anderbubble**
CONTRIBUTING file as mentioned in the [draft technical charter](https://docs.google.com/document/d/1SO6i_7of0BlSdajcFK7mvgMIVD9zmOP0n5KF4xEb2tk).

Open questions:

* Is the initial member list for the technical steering committee correct?
* Do we need to draft a code of conduct?
* Do we need signed-off and a DCO.md?
* We need a branching policy to reflect here
* Do the comments here about tests apply? Will they apply?

### CODE REVIEW

**Comment by anderbubble** [comment]
Added myself to the list as discussed in the community meeting.

File: CONTRIBUTING.md
Comment (elguero) [inline CONTRIBUTING.md:53]: Instead of saying essential, I would be specific.

"All bug fixes and regressions should be submitted against the development and main branch."
"All security fixes need to be submitted against the development branch, main branch and currently released tag"

File: CONTRIBUTING.md
Comment (elguero) [inline CONTRIBUTING.md:60]: I would add, before adding new functionality, check in with the community for feedback on your idea before embarking on a path that would not be accepted into the code base.  Something like that.

I do see it in the Getting Started section. It might be beneficial to repeat that idea here as well.  To be honest, I glossed over that point in Getting Started and jumped straight to here at first.

**Review by elguero — CHANGES_REQUESTED** [review]
A few comments on the document.

Plus, I would recommend that we establish a process for code reviews and merging.

We should establish how many code reviewers we need to approve PR.  I would recommend two approvers.

A PR should not be merged for at least two days.  For instance, since contributors are in different time zones, it would be nice to give the community a chance to review the code and make comments before it is merged.

Who is allowed to merge PR?  The TSC?  Maybe just make sure that is clear somewhere.

We should explain what our branching/merging strategy is.

Just some initial thoughts while reviewing this.

**Comment by anderbubble** [comment]
Thanks for the feedback, @elguero!

> I would recommend that we establish a process for code reviews and merging.

> We should establish how many code reviewers we need to approve PR. I would recommend two approvers.

> A PR should not be merged for at least two days.

> We should explain what our branching/merging strategy is.

So far, my goal has been to get a decision-making process in place, and then use that process to make such decisions. Anything that's in here already that goes beyond that is only there because it was in the template I was using, and I didn't see a need to remove stuff unless people wanted it removed.

I've taken these questions / suggestions and added them to a to-do list for the TSC, or whatever it ends up being.

> Who is allowed to merge PR? The TSC? Maybe just make sure that is clear somewhere.

My expectation is that this is just whomever is granted the ability to do that in GitHub, likely what the draft charter refers to as "committers." But this could / should be called out explicitly.

I'll respond to the others where they are.

File: CONTRIBUTING.md
Comment (anderbubble) [reply CONTRIBUTING.md:55]: This is a mistake. I thought I already updated this, but I'll do it now. Thanks for the catch.

Comment (anderbubble) [reply CONTRIBUTING.md]: Looks like I just hadn't pushed it.

File: CONTRIBUTING.md
Comment (anderbubble) [reply CONTRIBUTING.md]: I agree with making it more specific; but I think more discussion needs to be had about the branching strategy in order for that specificity to be correct. I'm afraid that the way people are currently talking about development vs main will lead to painful merges or (more like what's happening now) a divergence that makes one of them not as well used or tested. But rather than try to figure that out (or make my arguments) at this stage, I figured the lesser specificity (using the language from Apptainer) would make it easier to get something down, and then we could discuss the specifics next.

File: CONTRIBUTING.md
Comment (anderbubble) [reply CONTRIBUTING.md]: I just merged the Getting Started section into the PRs section, so there wouldn't be two places to look. ^_^

**Comment by anderbubble** [comment]
Since we're targeting a different set of bylaws, this is no longer relevant.

**Pull request closed without merging**
